EarlGrey/Action/GREYMultiFingerSwipeAction.m

```diff
@@ -0,0 +1,173 @@
+//
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "GREYMultiFingerSwipeAction.h"
+
+#import "Action/GREYPathGestureUtils.h"
+#import "Additions/NSError+GREYAdditions.h"
+#import "Additions/NSObject+GREYAdditions.h"
+#import "Additions/NSString+GREYAdditions.h"
+#import "Assertion/GREYAssertionDefines.h"
+#import "Assertion/GREYAssertions+Internal.h"
+#import "Common/GREYError.h"
+#import "Event/GREYSyntheticEvents.h"
+#import "Matcher/GREYAllOf.h"
+#import "Matcher/GREYMatcher.h"
+#import "Matcher/GREYMatchers.h"
+#import "Matcher/GREYNot.h"
+
+@implementation GREYMultiFingerSwipeAction {
+    /**
+     *  The direction in which the content must be scrolled.
+     */
+    GREYDirection _direction;
+    /**
+     *  The duration within which the swipe action must be complete.
+     */
+    CFTimeInterval _duration;
+    /**
+     *  Start point for the swipe specified as a percentage of the swipped element's accessibility frame.
+     */
+    CGPoint _startPercents;
+    /**
+     *  Number of parallel swipes
+     */
+    NSUInteger _numberOfFingers;
+}
+    
+- (instancetype)initWithDirection:(GREYDirection)direction
+                         duration:(CFTimeInterval)duration
+                  numberOfFingers:(NSUInteger)numberOfFingers
+                     percentPoint:(CGPoint)percents {
+    NSAssert(percents.x > 0.0f && percents.x < 1.0f,
+             @"xOriginStartPercentage must be between 0 and 1, exclusively");
+    NSAssert(percents.y > 0.0f && percents.y < 1.0f,
+             @"yOriginStartPercentage must be between 0 and 1, exclusively");
```

Comment:
You're using xOriginStartPercentage, but the actual value that you're checking is `percents.x` or `percents.y`. You should use:

@"percents.x must be between 0 and 1, exclusively"